server/src/main/java/org/elasticsearch/index/codec/bloomfilter/ES93BloomFilterStoredFieldsFormat.java

```diff
@@ -322,19 +312,245 @@ private void finishBloomFilterStoredFormat() throws IOException {
 
         @Override
         public int merge(MergeState mergeState) throws IOException {
-            // Skip merging the bloom filter for now
+            if (canOrBloomFilters(mergeState)) {
+                mergeBloomFiltersWithOr(mergeState);
+            } else {
+                rebuildBloomFilterFromSegments(mergeState);
+            }
             finishBloomFilterStoredFormat();
             return delegateWriter.merge(mergeState);
         }
 
+        private void mergeBloomFiltersWithOr(MergeState mergeState) throws IOException {
+            assert canOrBloomFilters(mergeState);
+
+            if (mergeState.storedFieldsReaders.length == 0) {
+                return;
+            }
+            assert mergeState.storedFieldsReaders[0] instanceof Reader;
+            Reader firstReader = (Reader) mergeState.storedFieldsReaders[0];
+            assert firstReader.bloomFilterFieldReader != null;
+
+            var mergedBloomFilterBitSetSizeInBits = firstReader.bloomFilterFieldReader.getBloomFilterBitSetSizeInBits();
+
+            var bloomFilterFieldInfo = mergeState.mergeFieldInfos.fieldInfo(bloomFilterFieldName);
+            maybeInitializeBloomFilterWriter(bloomFilterFieldInfo, mergedBloomFilterBitSetSizeInBits);
+            bloomFilterWriter.mergeBloomFiltersWithOr(mergeState);
+        }
+
+        private void rebuildBloomFilterFromSegments(MergeState mergeState) throws IOException {
+            final List<Fields> fields = new ArrayList<>();
+            final List<ReaderSlice> slices = new ArrayList<>();
+
+            int docBase = 0;
+
+            for (int readerIndex = 0; readerIndex < mergeState.fieldsProducers.length; readerIndex++) {
+                final FieldsProducer f = mergeState.fieldsProducers[readerIndex];
+
+                final int maxDoc = mergeState.maxDocs[readerIndex];
+                if (f != null) {
+                    // This can be somewhat expensive and we'll check the integrity again while merging the terms, should we skip it?
+                    f.checkIntegrity();
```

Comment:
I think it's OK to keep it: it should always be used with the synthetic _id postings formats which does nothing in `checkIntegrity`